portalnet/src/overlay_service.rs

```diff
@@ -3657,6 +3776,71 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn advance_find_content_query_with_connection_id() {
+        let mut service = task::spawn(build_service());
+
+        let (_, bootnode_enr) = generate_random_remote_enr();
+        let bootnode_node_id = bootnode_enr.node_id();
+        let bootnode_key = kbucket::Key::from(bootnode_node_id);
+
+        let data_radius = Distance::MAX;
+        let bootnode = Node {
+            enr: bootnode_enr.clone(),
+            data_radius,
+        };
+
+        let connection_direction = ConnectionDirection::Outgoing;
+        let status = NodeStatus {
+            state: ConnectionState::Connected,
+            direction: connection_direction,
+        };
+
+        let _ = service
+            .kbuckets
+            .write()
+            .insert_or_update(&bootnode_key, bootnode, status);
+
+        let target_content = NodeId::random();
+        let target_content_key = IdentityContentKey::new(target_content.raw());
+
+        let query_id = service.init_find_content_query(target_content_key, None, false);
+        let query_id = query_id.expect("Query ID for new find content query is `None`");
+
+        // update query in own span so mut ref is dropped after poll
```

Comment:
:+1: and this comment is more evidence to move away from `RwLock`, since it took me quite a while to discover that this lock wasn't being dropped and blocking the following code.